src/main/java/TaskList.java

```diff
@@ -0,0 +1,55 @@
+import tasks.Task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskList {
+
+    private List<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> taskList) {
+        this.taskList = new ArrayList<>(taskList);
+    }
+
+    public Task get(int index) {
+        return taskList.get(index);
+    }
+
+    public void addTask(Task t) {
+        taskList.add(t);
+    }
+
+    public Task deleteTask(int index) {
+        return taskList.remove(index);
+    }
+
+    public int size() {
+        return taskList.size();
+    }
+
+    public void printList() {
+        int ctr = 1;
+        for (Task task: taskList) {
+            System.out.println("\t" + ctr + "." + taskList.get(ctr - 1));
+            ctr++;
+        }
+    }
+
+    public TaskList findMatchingTasks(String keyword) {
```

Comment:
Minor point here, but I felt that the name `findMatchingTasks` might be better written as `findTasksThatMatchKeyword`. This is because `findMatchingTasks` makes me think that the code will find and group `Task`s based on their similarities, instead of a method that queries for `Task`s based on a given String.